src/main/java/com/textmessenger/service/MessageServiceJpa.java

```diff
@@ -0,0 +1,46 @@
+package com.textmessenger.service;
+
+import com.textmessenger.model.Message;
+import com.textmessenger.repository.MessageRepository;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class MessageServiceJpa implements MessageService {
+
+  private final MessageRepository messageRepository;
+
+  public MessageServiceJpa(MessageRepository messageRepository) {
+    this.messageRepository = messageRepository;
+  }
+
+  @Override
+  @Transactional
```

Comment:
If you have @Transactional annotation on a class level, you don't need it for each method